modal/_asgi.py

```diff
@@ -28,26 +33,32 @@ async def fetch_data_in():
             message_gen = function_io_manager.get_data_in.aio(function_call_id)
             try:
                 first_message = await asyncio.wait_for(message_gen.__anext__(), 5.0)
-            except asyncio.TimeoutError:
+            except (asyncio.TimeoutError, StopAsyncIteration):
                 if scope["type"] == "http":
-                    await messages_from_app.put({"type": "http.response.start", "status": 502})
                     await messages_from_app.put(
-                        {
-                            "type": "http.response.body",
-                            "body": b"Missing request, possibly due to cancellation or crash",
-                        }
+                        [
+                            {"type": "http.response.start", "status": 502},
+                            {
+                                "type": "http.response.body",
+                                "body": b"Missing request, possibly due to cancellation or crash",
+                            },
+                            {"type": "http.disconnect"},
+                        ]
                     )
-                    await messages_to_app.put({"type": "http.disconnect"})
                 elif scope["type"] == "websocket":
                     await messages_from_app.put(
-                        {
-                            "type": "websocket.close",
-                            "code": 1011,
-                            "reason": "Missing request, possibly due to cancellation or crash",
-                        }
+                        [
+                            {
+                                "type": "websocket.close",
+                                "code": 1011,
+                                "reason": "Missing request, possibly due to cancellation or crash",
+                            },
+                            {"type": "websocket.disconnect"},
+                        ]
                     )
-                    await messages_to_app.put({"type": "websocket.disconnect"})
```

Comment:
Oh wow, nice catch! Can't believe I didn't notice the disconnect was on a different channel, I kind of just mechanically moved all the "messages" to the same batch without looking to closely. Will add a test since it's quite bad this wasn't caught (even if it's an edge case/error state)